nim/tests/test_routes.py

```diff
@@ -1,103 +1,110 @@
+import json
 from nim.app import app
 import pytest
-import json
 
 
 @pytest.fixture
-def test_app():
+def client_app():
     return app.test_client()
 
 
-def test_index(test_app):
-    response = test_app.get('/', content_type='html/text')
+def assert_new_game_success(passed_client, req):
+    """
+    Asserts post to /new is 200
+    & state is not empty
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 200 and not len(json.loads(response.data)['state']) == 0
```

Comment:
Instead of `not x == 0` you should write `x != 0`.
But also, an empty list is False-y in Python, so you should write:
```
    assert response.status_code == 200 and json.loads(response.data)['state']
```
In this case though, we should check a bit more than that. We have `req` so we know how many piles there should be exactly, and we know the number of stones in each pile should be within a range.
So we should add that too.

Reply:
Okay 🙂 I'll add that